media/videostitcher/stitcher_test.go

```diff
@@ -404,3 +412,141 @@ func TestVodSessions(t *testing.T) {
+	}
+
+	// Get the live session.
+	testutil.Retry(t, 3, 2*time.Second, func(r *testutil.R) {
+		sessionName := fmt.Sprintf("projects/%s/locations/%s/liveSessions/%s", projectNumber, location, sessionID)
+		if err := getLiveSession(buf, tc.ProjectID, sessionID); err != nil {
+			r.Errorf("getLiveSession got err: %v", err)
+		}
+		if got := buf.String(); !strings.Contains(got, sessionName) {
+			r.Errorf("getLiveSession got\n----\n%v\n----\nWant to contain:\n----\n%v\n----\n", got, sessionName)
+		}
+	})
+	buf.Reset()
+
+	// No list or delete methods for live sessions
+
+	// Ad tag details
+
+	// To get ad tag details, you need to curl the main manifest and
+	// a rendition first. This supplies media player information to the API.
+	//
+	// Curl the playURI first. The last line of the response will contain a
+	// renditions location. Curl the live session name with the rendition
+	// location appended.
+
+	resp, err := http.Get(playURI)
+	if err != nil {
+		t.Errorf("\nError getting the play URI\n")
+	}
+	defer resp.Body.Close()
+	body, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		t.Errorf("ioutil.ReadAll: %v", err)
+	}
+
+	re = regexp.MustCompile(`renditions/.*`)
+	renditions := re.FindStringSubmatch(string(body))
+	if len(renditions) == 0 {
+		t.Errorf("\nRenditions not found in body: %s\n", string(body))
+	}
+
+	// playURI will be in the following format:
+	// https://videostitcher.googleapis.com/v1/projects/{project}/locations/{location}/liveSessions/{session-id}/manifest.m3u8?signature=...
+	// Replace manifest.m3u8?signature=... with the renditions location.
+
+	re = regexp.MustCompile(`manifest.m3u8.*`)
+	renditionURI := re.ReplaceAllString(playURI, renditions[0])
+	resp, err = http.Get(renditionURI)
+	if err != nil {
+		t.Errorf("\nError getting the rendition URI\n")
+	}
+	defer resp.Body.Close()
+
+	// List the ad tag details for a given live session.
+	adTagDetailsNamePrefix := fmt.Sprintf("projects/%s/locations/%s/liveSessions/%s/liveAdTagDetails/", projectNumber, location, sessionID)
+	if err := listLiveAdTagDetails(buf, tc.ProjectID, sessionID); err != nil {
+		t.Errorf("listLiveAdTagDetails got err: %v", err)
+	}
+	got = buf.String()
+
+	if !strings.Contains(got, adTagDetailsNamePrefix) {
+		t.Errorf("listLiveAdTagDetails got\n----\n%v\n----\nWant to contain:\n----\n%v\n----\n", got, adTagDetailsNamePrefix)
+	}
+	strSlice := strings.Split(got, "/")
+	adTagDetailsID := strSlice[len(strSlice)-1]
+	adTagDetailsID = strings.TrimRight(adTagDetailsID, "\n")
+	adTagDetailsName := fmt.Sprintf("projects/%s/locations/%s/liveSessions/%s/liveAdTagDetails/%s", projectNumber, location, sessionID, adTagDetailsID)
+	if !strings.Contains(got, adTagDetailsName) {
+		t.Errorf("listLiveAdTagDetails got\n----\n%v\n----\nWant to contain:\n----\n%v\n----\n", got, adTagDetailsName)
+	}
+	buf.Reset()
+
+	// Get the specified ad tag detail for a given live session.
+	testutil.Retry(t, 1, 2*time.Second, func(r *testutil.R) {
```

Comment:
I think setting maxAttempts to 1 means that it will not be retried. Consider using more generous limits to avoid test flakes.

Reply:
Changed to 3